hls4ml/templates/vivado/nnet_utils/nnet_pooling.h

```diff
@@ -27,15 +27,39 @@ template <int W, int N> ap_int<W> avg(ap_int<W> (&x)[N]) {
     return tmp;
 }
 
-template <int W, int I, int N> ap_fixed<W, I> avg(ap_fixed<W, I> (&x)[N]) {
+template <int W, int N> ap_int<W> avg(ap_uint<W> (&x)[N]) {
```

Comment:
didn't you mean to return `ap_uint<W>` here?

Reply:
Fixed